### PR TITLE
[extractor/doodstream] Fix download of not embedded video when the link is protected

### PR DESCRIPTION
**Pull request by Galiley**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

When the doodstream's video URL is not an embedded video (eg in the form of `https://dood.wf/e/<video_id>`), yt-dlp failed to download it. This PR should fix that issue

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/common.py [hunk 1]
```diff
@@ -543,9 +543,13 @@ def suitable(cls, url):
         # so that lazy_extractors works correctly
         return cls._match_valid_url(url) is not None
 
+    @classmethod
+    def _match_group(cls, url, group):
+        return cls._match_valid_url(url).group(group)
+
```
Comment (pukkandan) [inline yt_dlp/extractor/common.py:549]: unnecessary

File: yt_dlp/extractor/doodstream.py [hunk 2]
```diff
@@ -44,6 +44,10 @@ class DoodStreamIE(InfoExtractor):
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        if self._match_group(url, 'type') == 'd':
```
Comment (pukkandan) [inline yt_dlp/extractor/doodstream.py:47]:
```suggestion
        video_id, type_ = self._match_valid_url(url).group('id', 'type')
        if type_ == 'd':
```

**Review by pukkandan — CHANGES_REQUESTED** [review]
Also add test

File: yt_dlp/extractor/doodstream.py [hunk 3]
```diff
@@ -44,6 +44,10 @@ class DoodStreamIE(InfoExtractor):
...
+            webpage = self._download_webpage(url, None)
```
Comment (pukkandan) [inline yt_dlp/extractor/doodstream.py:48]:
```suggestion
            webpage = self._download_webpage(url, video_id)
```

File: yt_dlp/extractor/doodstream.py [hunk 4]
```diff
@@ -44,6 +44,10 @@ class DoodStreamIE(InfoExtractor):
...
+            video_id = self._html_search_regex(r'<iframe src=\"/e/([a-z0-9]+)\"', webpage, 'video_id')
```
Comment (pukkandan) [inline yt_dlp/extractor/doodstream.py:49]:
```suggestion
            video_id = self._html_search_regex(r'<iframe src="/e/(\w+)"', webpage, 'video id')
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Comment by Galiley** [comment]
@pukkandan I've tried to add a new test case to catch that particular issue but i've failed. I didn't use the one that required that fix because it's a piracy URL. However, i can provide you it in order to test on your own

**Comment by pukkandan** [comment]
If you cannot even give a non pirated example this should perhaps not be merged

**Comment by Galiley** [comment]
> If you cannot even give a non pirated example this should perhaps not be merged

Well the bug still exist whether or not the example is pirated

File: yt_dlp/extractor/doodstream.py [hunk 5]
```diff
@@ -52,7 +66,7 @@ def _real_extract(self, url):
         thumb = self._html_search_meta(['og:image', 'twitter:image'], webpage, default=None)
         token = self._html_search_regex(r'[?&]token=([a-z0-9]+)[&\']', webpage, 'token')
         description = self._html_search_meta(
-            ['og:description', 'description', 'twitter:description'], webpage, default=None)
+            ['og:description', 'description', 'twitter:description'], webpage, default=title)
```
Comment (pukkandan) [inline yt_dlp/extractor/doodstream.py:69]: why?

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/doodstream.py (hunk 5 above)
Comment (pukkandan) [reply yt_dlp/extractor/doodstream.py:69]: if there is no description it should be empty
```suggestion
            ['og:description', 'description', 'twitter:description'], webpage, default=None)
```

**Comment by navid-zamani** [comment]
The whole damn point of yt-dlp is to download videos from sites, which, according to the lies of the organized crime (aka pirates aka content mafia), is “piracy”.

yt-dlp literally exists, to circumvent the artificial scarcity and imaginary monopolism schemes of such criminals, especially when they are state-sponsored by rogue regimes that support them.

So what the hell is this??

This is like a VPN going “We’re blocking tianmen-square-trush.org, because China banned it”.
You might as well call the project dead by suicide after schizophrenia.

**Pull request closed without merging**